src/components/Guests.js

```diff
@@ -0,0 +1,86 @@
+import React, { Component } from "react";
+import styled from "@emotion/styled";
+import uuidv4 from "uuid/v4";
+
+import Guest from "./Guest";
+import GuestInput from "./GuestInput";
+
+const StyledGuestContainer = styled.div`
+  margin-top: 0.8em;
+`;
+
+const StyledGuestList = styled.div`
+  display: block;
+  border-radius: 0.1875em;
+  overflow: hidden;
+`;
+
+class Guests extends Component {
+  nameRef = null;
+
+  state = {
+    inputError: null
+  };
+
+  setInputRef = ref => {
+    this.nameRef = ref;
+  };
+
+  handleClearInput = () => {
+    this.nameRef.value = "";
+    this.setState({
+      inputError: null
+    });
+  };
+
+  handleAddGuest = () => {
+    const { handleAddGuest } = this.props;
+
+    if (this.nameRef.value) {
+      if ([...this.nameRef.value].length <= 42) {
+        let newGuest = { id: uuidv4(), name: this.nameRef.value };
+        handleAddGuest(newGuest);
+        this.handleClearInput();
+      } else {
+        this.setState({
+          inputError: "Limit exceeded"
+        });
+      }
+    } else {
+      this.setState({
+        inputError: "Field empty"
+      });
+    }
+  };
+
+  render() {
+    const { guests, handleRemoveGuest, maxGuests } = this.props;
+    const { inputError } = this.state;
+
+    return (
+      <StyledGuestContainer>
+        {guests.length > 0 ? (
+          <StyledGuestList>
+            {guests.map(guest => (
+              <Guest
+                key={`${guest.name}--${guest.id}`}
```

Comment:
Is there ever an instance where multiple `guests` have the same `id` value?

If not, then `guest.id` is more than sufficient as the `key` value.